### PR TITLE
Add configuration variables for trajectory checking utilities

### PR DESCRIPTION
**Pull request by gauravgarre**
## Description
Add configuration variables for trajectory checking utilities

## Associated Issue
Issue #1525 

## Design Documents
[Link](link-to-design-doc)

### Test Case 1
1. Step 1
2. Step 2
3. Step 3

Expected result: ???

### CODE REVIEW

**Comment by gauravgarre** [comment]
Pretty positive I did not do this right

File: soccer/src/soccer/planning/trajectory_utils.cpp [hunk 1]
```diff
@@ -28,8 +28,8 @@ bool trajectory_hits_static(const Trajectory& trajectory, const rj_geometry::Sha
     // something is probably wrong, but we'll still handle it (just scale dt
     // accordingly).
     // TODO(#1525): Make these config variables.
-    constexpr int kMaxIterations = 100;
-    constexpr RJ::Seconds kExpectedDt{0.05};
+    DECLARE_NS_INT64(kMaxIterations, int, 100);
```
Comment (kylestach) [inline soccer/src/soccer/planning/trajectory_utils.cpp]: You need to `DECLARE_` in the `.hpp` file and `DEFINE_` in the `.cpp` file.

Also, the syntax for this isn't quite right. The arguments of the macro are:
```cpp
DECLARE_NS_<INT64|FLOAT64|...>(kParameterModuleName, some::namespace::for::the::param, param_name);
```

File: soccer/src/soccer/planning/params.cpp [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#include <rj_param_utils/param.hpp>
+
+namespace trajectory_utils {
+    DEFINE_INT64(kMaxIterations, max_iterations, 100, "Limit on iterations")
```
Comment (kylestach) [inline soccer/src/soccer/planning/params.cpp:4]: I think it should be okay to put these directly in the `trajectory_utils.hpp/cpp` files.

File: soccer/src/soccer/planning/params.hpp [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include <rj_param_utils/param.hpp>
+
+namespace trajectory_utils {
+    constexpr auto kMaxIterations = "trajectory_utils";
+    constexpr auto kExpectedDt = "trajectory_utils";
```
Comment (kylestach) [inline soccer/src/soccer/planning/params.hpp:7]: These variables should be deleted. There should be a constant for `kPlanningParams` (the string constant is the module name).

File: soccer/src/soccer/planning/params.hpp [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+    DECLARE_INT64(kMaxIterations, max_iterations)
+    DECLARE_NS_FLOAT64(kExpectedDt, RJ::Seconds, expected_dt)
```
Comment (kylestach) [inline soccer/src/soccer/planning/params.hpp:10]: Don't make this a `_NS_` variable. You're already in an outer-level namespace.

**Review by kylestach — CHANGES_REQUESTED** [review]
Sorry for the late review! Here are some suggestions.

File: soccer/src/soccer/planning/trajectory_utils.cpp (hunk 1 above)
Comment (kylestach) [reply soccer/src/soccer/planning/trajectory_utils.cpp]: For reference the format for the non-namespaced version is just:
```cpp
DECLARE_<INT64|FLOAT64|...>(kParameterModuleName, param_name);
```

File: soccer/src/soccer/planning/params.hpp [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include <rj_param_utils/param.hpp>
+
+namespace trajectory_utils {
+    constexpr auto kMaxIterations = "trajectory_utils";
+    constexpr auto kExpectedDt = "trajectory_utils";
+
+    DECLARE_INT64(kMaxIterations, max_iterations)
```
Comment (mamarcus64) [inline soccer/src/soccer/planning/params.hpp:9]: not sure if I understand this correctly, but if kMaxIterations represents the same value as max_iterations why not just use max_iterations and cut out the middle-man variable

Comment (adithya-gv) [reply soccer/src/soccer/planning/params.hpp:9]: i think kMaxIterations is later modified to be a different value? but maybe im just understanding it differently as well.

Comment (kylestach) [reply soccer/src/soccer/planning/params.hpp:9]: This should be `DECLARE_INT(kPlanningParams, max_iterations)`.

The first argument to this macro is a string representing the "module name", which is just how we group together a bunch of parameters. The second one is the variable name of the parameter, which can be given an initial value but is also able to be changed through ROS.

**Pull request closed without merging**